installer/scripts/service_control

```diff
@@ -45,6 +46,13 @@ is_omsagent_running()
     fi
 }
 
+warn_if_agent_not_onboarded()
+{
+    if [ ! -f $OMSADMIN_CONF ]; then
+       echo "Warning: Agent is not onboarded. omsagent cannot be registered as a service."
+    fi
+}
```

Comment:
I think a warning here is not enough. If the agent isn't onboarded, you should not continue with code execution.

Thus, you should return a value here and, when calling, abort code execution if we're not onboarded.